### ISSUE TITLE
vmstorage starts merging indexdb parts even if there is not enough disk space 

### ISSUE THREAD

**Issue by turbotankist** [VictoriaMetrics/VictoriaMetrics#1560]
**Describe the bug**
vmstorage starts merging indexdb parts even if there is not enough disk space. Then pod with vmstorage restarts and tmp directory is cleared. The process starts from the beginning. 
![image](https://user-images.githubusercontent.com/9462236/130589167-b48f7441-8b3c-4f5b-8386-081925e34310.png).
```
/ # df -h
Filesystem                Size      Used Available Use% Mounted on
overlay                 200.0G      7.5G    192.5G   4% /
tmpfs                    64.0M         0     64.0M   0% /dev
tmpfs                    29.9G         0     29.9G   0% /sys/fs/cgroup
/dev/xvdbs              984.2G    978.6G      5.5G  99% /storage
/dev/xvda1              200.0G      7.5G    192.5G   4% /dev/termination-log
/dev/xvda1              200.0G      7.5G    192.5G   4% /etc/resolv.conf
/dev/xvda1              200.0G      7.5G    192.5G   4% /etc/hostname
/dev/xvda1              200.0G      7.5G    192.5G   4% /etc/hosts
shm                      64.0M         0     64.0M   0% /dev/shm
tmpfs                    29.9G     12.0K     29.9G   0% /run/secrets/kubernetes.io/serviceaccount
tmpfs                    29.9G         0     29.9G   0% /proc/acpi
tmpfs                    64.0M         0     64.0M   0% /proc/kcore
tmpfs                    64.0M         0     64.0M   0% /proc/keys
tmpfs                    64.0M         0     64.0M   0% /proc/latency_stats
tmpfs                    64.0M         0     64.0M   0% /proc/timer_list
tmpfs                    64.0M         0     64.0M   0% /proc/sched_debug
tmpfs                    29.9G         0     29.9G   0% /proc/scsi
tmpfs                    29.9G         0     29.9G   0% /sys/firmware
```
```
# du -sh /storage/*
647.2M	/storage/cache
216.1G	/storage/data
0	/storage/flock.lock
762.9G	/storage/indexdb
16.0K	/storage/lost+found
8.0K	/storage/metadata
4.0K	/storage/snapshots
```
```
 # du -sh /storage/indexdb/1671AD0C9B9502F4/*
26.9M	/storage/indexdb/1671AD0C9B9502F4/1055440_4090_169DD4072039F563
3.6M	/storage/indexdb/1671AD0C9B9502F4/118667_448_169DD4072039F5BA
380.2M	/storage/indexdb/1671AD0C9B9502F4/13494147_54744_169DD4072039F126
4.1G	/storage/indexdb/1671AD0C9B9502F4/140014711_571801_169DD4072039C668
39.1M	/storage/indexdb/1671AD0C9B9502F4/1497938_5814_169DD4072039F46C
24.0K	/storage/indexdb/1671AD0C9B9502F4/14_1_169DD4072039E3A0
487.7M	/storage/indexdb/1671AD0C9B9502F4/17427764_71781_169DD4072039DF55
57.7M	/storage/indexdb/1671AD0C9B9502F4/2160102_8478_169DD4072039F2BB
52.0K	/storage/indexdb/1671AD0C9B9502F4/2195_5_169DD4072039F5C4
808.0K	/storage/indexdb/1671AD0C9B9502F4/31654_108_169DD4072039F5C1
96.0K	/storage/indexdb/1671AD0C9B9502F4/4364_11_169DD4072039F5C2
128.0K	/storage/indexdb/1671AD0C9B9502F4/4961_14_169DD4072039F5C3
18.9M	/storage/indexdb/1671AD0C9B9502F4/558406_2129_169DD4072039F595
32.0K	/storage/indexdb/1671AD0C9B9502F4/607_2_169DD4072039F5B8
208.1G	/storage/indexdb/1671AD0C9B9502F4/6400983426_27476158_169CF1C36700AC85
28.0K	/storage/indexdb/1671AD0C9B9502F4/646_2_169DD4072039F5B9
32.0K	/storage/indexdb/1671AD0C9B9502F4/686_2_169DD4072039F5BE
227.9G	/storage/indexdb/1671AD0C9B9502F4/7516758288_32516181_168CBE80303BF42B
4.0K	/storage/indexdb/1671AD0C9B9502F4/converted-to-v1.28.0
0	/storage/indexdb/1671AD0C9B9502F4/flock.lock
318.7G	/storage/indexdb/1671AD0C9B9502F4/tmp      <-- merging
4.0K	/storage/indexdb/1671AD0C9B9502F4/txn
```

metrics from victoriametrics dashboard
![image](https://user-images.githubusercontent.com/9462236/130591832-e614981f-fe35-488d-a989-6fd4cb37faab.png)

![image](https://user-images.githubusercontent.com/9462236/130590418-e0ee3640-a267-49c2-94bc-401fb330c15d.png)
![image](https://user-images.githubusercontent.com/9462236/130590532-29a9c377-672f-4a7a-b24c-dd662ffdb9c5.png)
![image](https://user-images.githubusercontent.com/9462236/130591314-ee081e84-6fd3-4190-ad0c-592dc5d6e4d8.png)

**Version**
```
./vmstorage-prod --version
vmstorage-20210819-093427-tags-v1.64.1-cluster-0-gee1f3414d
```
and 1.63.0 had the same problem

**Used command-line flags**
```
8-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:12	build version: vmstorage-20210819-093427-tags-v1.64.1-cluster-0-gee1f3414d
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:13	command line flags
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "bigMergeConcurrency"="0" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "dedup.minScrapeInterval"="1m0s" (is_set=true)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "denyQueriesOutsideRetention"="false" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "enableTCP6"="false" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "envflag.enable"="false" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "envflag.prefix"="" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "finalMergeDelay"="30m0s" (is_set=true)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "forceFlushAuthKey"="secret" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "forceMergeAuthKey"="secret" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "fs.disableMmap"="false" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "http.connTimeout"="2m0s" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "http.disableResponseCompression"="false" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "http.idleConnTimeout"="1m0s" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "http.maxGracefulShutdownDuration"="7s" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "http.pathPrefix"="" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "http.shutdownDelay"="0s" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "httpListenAddr"=":8482" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "logNewSeries"="false" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "loggerDisableTimestamps"="false" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "loggerErrorsPerSecondLimit"="0" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "loggerFormat"="default" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "loggerLevel"="INFO" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "loggerOutput"="stderr" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "loggerTimezone"="UTC" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "loggerWarnsPerSecondLimit"="0" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "memory.allowedBytes"="0" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "memory.allowedPercent"="60" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "precisionBits"="64" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "retentionPeriod"="4" (is_set=true)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "rpc.disableCompression"="false" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "search.maxTagKeys"="secret" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "search.maxTagValueSuffixesPerSearch"="100000" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "search.maxTagValues"="100000" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "search.maxUniqueTimeseries"="40000000" (is_set=true)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "smallMergeConcurrency"="0" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "snapshotAuthKey"="secret" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "storage.maxDailySeries"="0" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "storage.maxHourlySeries"="0" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "storageDataPath"="/storage" (is_set=true)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "tls"="false" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "tlsCertFile"="" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "tlsKeyFile"="secret" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "version"="false" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "vminsertAddr"=":8400" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/lib/logger/flag.go:28	flag "vmselectAddr"=":8401" (is_set=false)
2021-08-23T11:13:47.786Z	info	VictoriaMetrics/app/vmstorage/main.go:65	opening storage at "/storage" with -retentionPeriod=4
```

**Comment by valyala** [comment]
@turbotankist , could you build `vmstorage` component from the commit b885bd9b7dcd1b754a096a45407b2cacaef9ecc4 and verify whether it doesn't start merging big indexdb parts if there is no enough free disk space? See [build instructions for cluster components of VictoriaMetrics](https://docs.victoriametrics.com/Cluster-VictoriaMetrics.html#building-from-sources). It is OK to build only `vmstorage` component, while using other components - vminsert and vmselect - from v1.64.1.

**Comment by turbotankist** [comment]
we already expanded disk space and need to wait a time for a next merging

**Comment by valyala** [comment]
The improved logic for detecting free disk space has been included in VictoriaMerics [v1.65.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing this issue as fixed.

@turbotankist , feel free re-opening the issue if it isn't fixed for your setup.
